packages/babel-plugin-effects/src/handler-method-visitor.ts

```diff
@@ -117,9 +117,13 @@ const makeHandlerMethod = (
       const declaration = binding?.path.find(
         x => types.isVariableDeclaration(x) || types.isFunctionDeclaration(x)
       );
+
       if (declaration) {
         callExpressionDeclarations.add(declaration.node);
-        markPathForRemoval(declaration);
+
+        // Not the most elegant.
+        declaration.parent?.parent?.type === "Program" &&
+          markPathForRemoval(declaration);
```

Comment:
here's what i was looking for. what types of decls get pruned?

Reply:
Yeah, so... should we even prune at all? If we don't, there's some dead code that will be laying around.

When we're creating a handler, there is kind of a restriction at this point that `recall`s must be within the same scope of a `handle`r block. If functions are called outside of the handler block, they're hoisted inside of the handle. This obviously wouldn't work for an imported function. Hard requirement right now. 

Maybe the transform plugin should throw if that's the case?

Anyways, to answer your question, we prune a declaration if it's referenced in a handler and it's at the top-level of the program, _after_ the entire transform has been performed.

Reply:
We could get around this restriction by currying any function that recalls 🤔. The transformation would look like this:

```javascript
const avgJoeFn = () => {
   recall 'great.'
};
```
↓↓↓
```javascript
const wrappedAvgJoeFn = (data, resume) => function*(){
  yield resume('great')
}
```

And handlers would then call as such

```javascript
{
[myGreatHandler]*(e, resume){
   // other generated boiler plate
  return yield wrappedAvgJoeFn(e, resume)()
}
}
```